quinn-proto/src/crypto.rs

```diff
@@ -59,6 +60,29 @@ impl TlsSession {
 }
 
 impl CryptoSession for TlsSession {
+    fn alert(&self) -> Option<AlertDescription> {
+        self.get_alert()
+    }
+
+    fn alpn_protocol(&self) -> Option<&[u8]> {
+        self.get_alpn_protocol()
+    }
+
+    fn early_secret(&self) -> Option<&[u8]> {
+        self.get_early_secret()
+    }
+
+    fn is_handshaking(&self) -> bool {
+        match self {
+            TlsSession::Client(session) => session.is_handshaking(),
+            TlsSession::Server(session) => session.is_handshaking(),
+        }
```

Comment:
Why does this explicitly destructure while the other wrappers go through the `Deref` impl?

Reply:
You could probably make rustc happy with UFCS, but yeah, not important for short-lived code.